tests/integration/test_propagation.py

```diff
@@ -68,16 +69,36 @@ def downstream_tracer():
     tracer.shutdown()
 
 
+CONTEXT_HEADERS = {
+    "x-datadog-trace-id": "1234",
+    "x-datadog-parent-id": "5678",
+    "x-datadog-sampling-priority": "1",
+    "x-datadog-tags": "_dd.p.dm=-1",
+}
+
+
 @pytest.mark.snapshot()
 def test_sampling_decision_downstream(downstream_tracer):
-    headers = {
-        "x-datadog-trace-id": "1234",
-        "x-datadog-parent-id": "5678",
-        "x-datadog-sampling-priority": "1",
-        "x-datadog-tags": "_dd.p.dm=-1",
-    }
-    context = HTTPPropagator.extract(headers)
+    context = HTTPPropagator.extract(CONTEXT_HEADERS)
     downstream_tracer.context_provider.activate(context)
 
     with downstream_tracer.trace("p", service="downstream") as span:
         span.set_tag(MANUAL_DROP_KEY)
+
+
+@pytest.fixture
+def tracer_with_single_span_sampling_enabled():
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0}]')):
```

Comment:
I think you want this configuration instead: https://docs.datadoghq.com/tracing/trace_pipeline/ingestion_mechanisms/?tab=python&code-lang=python#single-spans

E_TOO_MANY_SAMPLING_OPTIONS